src/main/scala/io/iohk/ethereum/db/components/StoragesComp.scala

```diff
@@ -0,0 +1,25 @@
+package io.iohk.ethereum.db.components
+
+import io.iohk.ethereum.db.storage._
+
+trait StoragesComp {
+
+  def storages: Storages
+
+  trait Storages {
+
+    def blockHeadersStorage: BlockHeadersStorage
```

Comment:
You probably wouldn't write 
`val i = 5`
`def j = 5`
blockHeadersStorage is a reference to one instance of a hopefully immutable Bloc...Storage. (<- if this is not true, then ignore the rest) It's not a Block...Storage generator. You are depending on implementation of the trait to override it with a val to make it correct. If you leave it as a def *and* implement that with another trait that is also a 'def' when it's trying to express a val you could end up with an instance per reference... I suggest the intent is  clearer using `val`...